modules/document_repository/php/document_repository.class.inc

```diff
@@ -407,6 +407,16 @@ class Document_Repository extends \NDB_Menu_Filter
                 =& $this->tpl_data['File_categories'][$item['File_category']];
             $DisplayItem  = $item;
             $DisplayItem['File_name'] = utf8_encode($item['File_name']);
+            $DisplayItem['Data_dir']  = strtr(
+                rawurlencode($DisplayItem['Data_dir']),
+                array(
+                 '%21' => '!',
+                 '%2A' => '*',
+                 '%27' => "'",
+                 '%28' => '(',
+                 '%29' => ')',
+                )
```

Comment:
Just a question. Does this represent all the characters list that need to be encoded?
Are space and % sign accepted in files name?

Reply:
No,
rawurlencode — URL-encode according to RFC 3986 and strtr — Translate characters or replace substrings. The section you referenced of the array is to make it equivalent to the javascript function encodeURIComponent() in image uploading.

Reply:
I'm a little confused about the relationship between this (which doesn't involve "+") and the PR title/description (which just says it's fixing "+"?

Reply:
@driusan The rawurlencode function handles the "+" character(s) conversion and many other characters needed to be converted as the function is designed for the problem the PR is designed to fix. The strtr function handles replacing what is in the array that @PapillonMcGill quoted for the output of the rawurlencode function and to be equivalent to how the javascript function handles conversion. The reason for it all is to make the javascript & php output the same.

Reply:
Can you add that explanation to a code comment so that the reasoning isn't lost to someone reading it in the future?